csh_auth.go

```diff
@@ -0,0 +1,156 @@
+package csh_auth
+
+import (
+    "fmt"
+    "os"
+    "net/http"
+    "time"
+    log "github.com/sirupsen/logrus"
+    jwt "github.com/dgrijalva/jwt-go"
+    oidc "github.com/coreos/go-oidc"
+    "golang.org/x/net/context"
+    "golang.org/x/oauth2"
+    "github.com/gin-gonic/gin"
+)
+
+
+var clientID = os.Getenv("csh_auth_client_id")
```

Comment:
gin supports the idea of middleware components which can be instantiated by the user. I'd recommend looking through that. You can create a middlewhere constructor that pulls all this data into a config that way all the globals are removed.